### DISCUSSION TITLE
Is the SVG file a definitive solution or how to?

### DISCUSSION THREAD

**Discussion by Trzinka** [ExperienceLovelace/ha-floorplan#427] [Q&A] [answered]
## Hi!

# I still haven't found a solution how to display different data/entities in the floor plan card!

**Some time ago I had the same problem, which I somehow solved (display on the tablet), but now I would like to see the same thing on my mobile phone:**

`Appearance on a computer monitor:`

![20240214-FP-Frames](https://github.com/ExperienceLovelace/ha-floorplan/assets/40424965/3c35c1bb-5ff7-4fdc-bfab-b699561c928f)

`Appearance on a tablet:`

![Screenshot_20240214-181541](https://github.com/ExperienceLovelace/ha-floorplan/assets/40424965/2cfc6d88-dd60-46b1-8d67-858ad34ff7a5)

`And appearance on a mobile phone:`

![Mobitel_Home Assistant](https://github.com/ExperienceLovelace/ha-floorplan/assets/40424965/6985b12a-961f-4807-97f6-028f79da6c55)

**The problem arises, even though I mostly use viewport height or viewport width (vh/vw) in yaml and css, I can't manage to set the appropriate ratio so that it looks at least approximately the same on all devices.**

# I thought for a bit and came to the conclusion that apparently the floor plan card itself can take care of appropriate display on different devices (fewer settings are required), so I am interested in whether it is possible and how to add space to the SVG file where I would insert the cards. I have the current settings at the link https://github.com/ExperienceLovelace/ha-floorplan/discussions/411 in a zip file! 

`The card display is as follows:`

![bitmap](https://github.com/ExperienceLovelace/ha-floorplan/assets/40424965/605599c1-1267-423e-91c8-dff5424ea4f3)

**Comment by OzGav** [comment]
Personally i would make the whole thing a floorplan. The only thing I haven’t done before is that event list but I think it wouldn’t be too hard. Look in the discussions where I showed how to create the weather card in floorplan. The cameras might be more difficult but you could always try using the sidebar view and put them there.

**Reply by Trzinka** [reply]
**Hi!** 

# Thanks for the reply! Could you please also attach:

 `floorplan.css` 

and `/local/icons/weather_icons/animated/${entity.state}.svg` 

and probably also `/local/icons/weather_icons/animated/' + imageName + '.svg'`

**Reply by Trzinka** [reply]
`And another question? Do you have a weather card built into the floor plan?`

## By this I mean do you have an svg file that combines the weather card and the floor plan card or, if you do it separately, I would ask you for the `yaml file`

**Reply by OzGav** [reply]
See here https://github.com/ExperienceLovelace/ha-floorplan/discussions/324

**Reply by Trzinka** [reply]
I checked the page you mention in the link. While looking through the yaml code that is in the zip file next to the svg file I noticed links to the aforementioned files that are not in the zip file.

**Reply by OzGav** [reply]
> Do you have a weather card built into the floor plan?

I don’t but I just showed how it could be done. In your case you would open your SVG in Inkscape, expand the left side and just add the weather card there. I would expand the top and add the person state as well. Just make it one big floorplan and then it will sort out the sizing on different devices for you.

**Reply by OzGav** [reply]
From memory I don’t think the css file is required as I don’t apply any classes. You can find weather icons in various places

edit: some icons are here https://github.com/DavidFW1960/bom-weather-card

**Reply by Trzinka** [reply]
# Thank you. 

**I was also thinking in that direction. I have another question about persons, in all probability I have to make a square for the image for the person and below it for the text, but how do you make a square for the icon or is it the same picture?**

**Reply by OzGav** [reply]
I would just have the home picture in the bottom layer and then put the away picture on top. Then just show or hide the away picture according to the state of the entity.

**Reply by Trzinka** [reply]
## Do I understand correctly that I cannot use other cards from HACS in the floor plan card?

**If I want to imitate a card, I have to "draw" it in an svg file and then call entity state!**

**Reply by OzGav** [reply, marked as answer]
Correct. There is no way to overlay or embed any other HA card into ha-floorplan. You should just recreate the card yourself as I did with the weather card. You can of course call up the more-info with a tap action, I do this for the air conditioner

**Reply by Trzinka** [reply]
# Thank you, I got it. 

`I have a lot of work to do now obviously!`
